hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/oauth2/TestWorkloadIdentityTokenProvider.java

```diff
@@ -77,4 +83,45 @@ public void testTokenDoesNotExpireTooSoon() {
 
     assertFalse(provider.hasEnoughTimeElapsedSinceLastRefresh());
   }
+
+  /**
+   * Test that the correct token is read from the token file.
+   *
+   * @throws IOException if the token file is empty or file I/O fails.
+   */
+  @Test
+  public void testGetToken() throws IOException {
+    long startTime = System.currentTimeMillis();
+    File tokenFile = File.createTempFile("azure-identity-token", "txt");
+    FileUtils.write(tokenFile, TOKEN, StandardCharsets.UTF_8);
+    AzureADToken azureAdToken = new AzureADToken();
+    WorkloadIdentityTokenProvider tokenProvider = Mockito.spy(
+        new WorkloadIdentityTokenProvider(AUTHORITY, TENANT_ID, CLIENT_ID, tokenFile.getPath()));
+    Mockito.doReturn(azureAdToken)
+        .when(tokenProvider).getTokenUsingJWTAssertion(TOKEN);
+    assertEquals(azureAdToken, tokenProvider.getToken());
+    assertTrue("token fetch time was not set correctly", tokenProvider.getTokenFetchTime() > startTime);
+  }
+
+  /**
+   * Test that an exception is thrown when the token file is empty.
+   *
+   * @throws IOException if file I/O fails.
+   */
+  @Test
+  public void testGetTokenThrowsWhenClientAssertionIsEmpty() throws IOException {
+    File tokenFile = File.createTempFile("azure-identity-token", "txt");
+    AzureADToken azureAdToken = new AzureADToken();
+    WorkloadIdentityTokenProvider tokenProvider = Mockito.spy(
+        new WorkloadIdentityTokenProvider(AUTHORITY, TENANT_ID, CLIENT_ID, tokenFile.getPath()));
+    Mockito.doReturn(azureAdToken)
+        .when(tokenProvider).getTokenUsingJWTAssertion(TOKEN);
+    boolean exception = false;
```

Comment:
Use LambdaTestUtils.intercept() here, ideally looking for the error string as well as exception class.